<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>PubForm</class>
 <widget class="QWidget" name="PubForm">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>540</width>
    <height>222</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <layout class="QHBoxLayout" name="horizontalLayout" stretch="1,0">
   <property name="leftMargin">
    <number>0</number>
   </property>
   <item>
    <widget class="QGroupBox" name="groupBox">
     <property name="title">
      <string/>
     </property>
     <layout class="QGridLayout" name="gridLayout">
      <item row="1" column="2">
       <widget class="QComboBox" name="qOSComboBox">
        <item>
         <property name="text">
          <string>QOS0</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>QOS1</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>QOS2</string>
         </property>
        </item>
       </widget>
      </item>
      <item row="0" column="0">
       <widget class="QLabel" name="topicLabel">
        <property name="text">
         <string>Topic(*):</string>
        </property>
       </widget>
      </item>
      <item row="6" column="0">
       <widget class="QLabel" name="messageLabel">
        <property name="text">
         <string>Message(*):</string>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="qOSLabel">
        <property name="text">
         <string>QOS:</string>
        </property>
       </widget>
      </item>
      <item row="6" column="2" colspan="5">
       <widget class="QTextEdit" name="payloadEdit"/>
      </item>
      <item row="0" column="2" colspan="5">
       <widget class="QLineEdit" name="topicLineEdit"/>
      </item>
      <item row="2" column="0">
       <widget class="QLabel" name="retainLabel">
        <property name="text">
         <string>Retain:</string>
        </property>
       </widget>
      </item>
      <item row="2" column="2">
       <widget class="QCheckBox" name="retainCheckBox"/>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <layout class="QVBoxLayout" name="verticalLayout">
     <item>
      <widget class="QPushButton" name="pubButton">
       <property name="text">
        <string>&amp;Publish</string>
       </property>
      </widget>
     </item>
     <item>
      <spacer name="verticalSpacer">
       <property name="orientation">
        <enum>Qt::Vertical</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>20</width>
         <height>40</height>
        </size>
       </property>
      </spacer>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <tabstops>
  <tabstop>topicLineEdit</tabstop>
  <tabstop>qOSComboBox</tabstop>
  <tabstop>retainCheckBox</tabstop>
  <tabstop>payloadEdit</tabstop>
  <tabstop>pubButton</tabstop>
 </tabstops>
 <resources/>
 <connections>
  <connection>
   <sender>pubButton</sender>
   <signal>clicked()</signal>
   <receiver>PubForm</receiver>
   <slot>onPublish()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>507</x>
     <y>23</y>
    </hint>
    <hint type="destinationlabel">
     <x>539</x>
     <y>104</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>topicLineEdit</sender>
   <signal>textEdited(QString)</signal>
   <receiver>PubForm</receiver>
   <slot>onPublishReady()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>193</x>
     <y>38</y>
    </hint>
    <hint type="destinationlabel">
     <x>435</x>
     <y>3</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>payloadEdit</sender>
   <signal>textChanged()</signal>
   <receiver>PubForm</receiver>
   <slot>onPublishReady()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>371</x>
     <y>152</y>
    </hint>
    <hint type="destinationlabel">
     <x>446</x>
     <y>252</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>onPublishReady()</slot>
  <slot>onPublish()</slot>
 </slots>
</ui>
